apps/angular/anchor-scrolling/src/app/home.component.ts

```diff
@@ -1,20 +1,34 @@
-import { Component } from '@angular/core';
+import { ViewportScroller } from '@angular/common';
+import { Component, inject } from '@angular/core';
+import { RouterModule } from '@angular/router';
 import { NavButtonComponent } from './nav-button.component';
 
 @Component({
   standalone: true,
-  imports: [NavButtonComponent],
+  imports: [NavButtonComponent, RouterModule],
   selector: 'app-home',
   template: `
-    <nav-button href="/foo" class="fixed left-1/2 top-3">Foo Page</nav-button>
-    <div id="top" class="h-screen bg-gray-500">
+    <nav-button [routerLink]="'/foo'" class="fixed left-1/2 top-3">
+      Foo Page
+    </nav-button>
+    <div id="top" class="h-screen bg-gray-500" (click)="scrollTo('bottom')">
       Empty
-      <nav-button href="#bottom">Scroll Bottom</nav-button>
+      <nav-button>Scroll Bottom</nav-button>
     </div>
-    <div id="bottom" class="h-screen bg-blue-300">
+    <div id="bottom" class="h-screen bg-blue-300" (click)="scrollTo('top')">
       I want to scroll each
-      <nav-button href="#top">Scroll Top</nav-button>
+      <nav-button>Scroll Top</nav-button>
     </div>
   `,
 })
-export class HomeComponent {}
+export class HomeComponent {
+  scroller = inject(ViewportScroller);
+
+  scrollTo(anchor: string) {
```

Comment:
You have the same code for `scrollTo` in `FooComponent`, you should refactor this in a service for example.

Reply:
here also removed unused scroller

Reply:
Add service and scrolling function moved to service file